crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs

```diff
@@ -304,3 +297,91 @@ pub(crate) fn filter_contained(edits: Vec<Edit>) -> Vec<Edit> {
     }
     filtered
 }
+
+#[derive(Copy, PartialEq, Clone)]
+enum State {
+    Literal,
+    Annotated,
+    AnnotatedNonFirstElm,
```

Comment:
I think I would switch this around and have `AnnotatedFirst` and `AnnotatedRest` or something similar to make it explicit about the state.